crates/cairo-lint-core/src/lints/bitwise_for_parity_check.rs

```diff
@@ -0,0 +1,21 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_syntax::node::ast::{BinaryOperator, ExprBinary};
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::TypedSyntaxNode;
+
+pub const BITWISE_FOR_PARITY: &str =
+    "You seem to be trying to use `&` for parity check. Consider using `DivRem::div_rem()` instead.";
+
+pub fn check_bitwise_for_parity(db: &dyn SyntaxGroup, node: &ExprBinary, diagnostics: &mut Vec<PluginDiagnostic>) {
```

Comment:
You should check that this type indeed implements `DivRem` so i don't think you should be using `SyntaxGroup` but use `SemanticGroup`